app/assets/javascripts/short-form/ShortFormNavigationService.js.coffee

```diff
@@ -1,61 +1,147 @@
-    'name': {callback: ['checkAfterNamePage']}
-    'contact': {callback: ['checkIfAddressVerificationNeeded', 'checkPreferenceEligibility']}
-    'verify-address': {path: 'alternate-contact-type', callback: ['checkPreferenceEligibility']}
-    'alternate-contact-type': {callback: ['checkIfAlternateContactInfoNeeded']}
-    'alternate-contact-name': {path: 'alternate-contact-phone-address'}
-    'alternate-contact-phone-address': {callback: ['goToLandingPage'], params: 'Household'}
+    'prerequisites':
+      callbacks: [
+        Service.goToAndTrackFormSuccess.bind(null, 'dahlia.short-form-application.name')
+      ]
+    'name':
+      scopedCallbacks: [{func: 'checkAfterNamePage'}]
+    'contact':
+      scopedCallbacks: [
+        {func: 'checkIfAddressVerificationNeeded'}
+        {func: 'checkPreferenceEligibility'}
+      ]
+    'verify-address':
+      scopedCallbacks: [{func: 'checkPreferenceEligibility'}]
+      path: 'alternate-contact-type'
+    'alternate-contact-type':
+      scopedCallbacks: [{func: 'checkIfAlternateContactInfoNeeded'}]
+    'alternate-contact-name':
+      path: 'alternate-contact-phone-address'
+    'alternate-contact-phone-address':
+      callbacks: [Service.goToLandingPage.bind(null, 'Household')]
     # household
-    'household-intro': {callback: ['validateHouseholdEligibility'], params: 'householdMatch'}
-    'household-members': {callback: ['validateHouseholdEligibility'], params: 'householdMatch'}
-    'household-member-form': {callback: ['addHouseholdMember', 'checkPreferenceEligibility']}
-    'household-member-form-edit': {callback: ['addHouseholdMember', 'checkPreferenceEligibility']}
-    'household-member-verify-address': {path: 'household-members', callback: ['checkPreferenceEligibility']}
-    'household-public-housing': {callback: ['checkIfPublicHousing']}
-    'household-monthly-rent': {callback: ['checkIfReservedUnits']}
-    'household-reserved-units-veteran': {callback: ['checkIfReservedUnits'], params: RESERVED_TYPES.DISABLED}
-    'household-reserved-units-disabled': {path: 'household-priorities'}
+    'household-intro':
+      scopedCallbacks: [{
+        func: 'validateHouseholdEligibility'
+        param: 'householdMatch'
+      }]
+    'household-members':
+      scopedCallbacks: [{
+        func: 'validateHouseholdEligibility'
+        param: 'householdMatch'
+      }]
+    'household-member-form':
+      scopedCallbacks: [
+        {func: 'addHouseholdMember'}
+        {func: 'checkPreferenceEligibility'}
+      ]
+    'household-member-form-edit':
+      scopedCallbacks: [
+        {func: 'addHouseholdMember'}
+        {func: 'checkPreferenceEligibility'}
+      ]
+    'household-member-verify-address':
+      scopedCallbacks: [{func: 'checkPreferenceEligibility'}]
+      path: 'household-members'
+    'household-public-housing': {scopedCallbacks: [{func: 'checkIfPublicHousing'}]}
+    'household-monthly-rent': {scopedCallbacks: [{func: 'checkIfReservedUnits'}]}
```

Comment:
isn't HH monthly rent just if rent burdened is a pref? why is it checking for reserved units?

Reply:
@akegan and i discussed this over slack call and went over how the short form navigation checks whether it needs to display pages for reserved units